engine/gui/widgets.py

```diff
@@ -106,10 +108,86 @@ def create(
                 None,
             )
 
-        return GUISpec(
+        gui_spec = GUISpec(
             assets=[Asset(**a) for a in spec.get("assets", [])],
             buttons=buttons,
             images=[Image(**b) for b in spec.get("images", [])],
             cancel_action=cancel_action,
             initial_selected_button=selected_button,
         )
+
+        cls.validate(gui_spec)
+        return gui_spec
+
+    def validate(self) -> None:
+        """Validates the GUISpec."""
+
+        # Validation errors for images have not been tested
+        if not self.assets:
+            raise ValidationError("Must have at least one asset")
+        if not self.buttons and not self.images:
+            raise ValidationError("Must have at least one button or image")
+
+        duplicate_assets = [
+            asset for asset, count in Counter(self.assets).items() if count > 1
+        ]
+        if len(duplicate_assets) > 0:
+            raise ValidationError(f"Duplicate asset name(s): {duplicate_assets}")
+
+        if len(set(self.assets)) != len(set(self.buttons)) + len(set(self.images)):
+            # Can be general with the commented error below or
+            # more detailed by iterating through the assets.
+            # raise ValidationError(
+            #     "Number of assets does not match the number of buttons and/or images."
+            # )
+            for asset in self.assets:
+                if self.buttons and asset.name not in [
+                    a.selected_image_asset for a in self.buttons
+                ]:
+                    raise ValidationError(f"Asset name not in buttons: {asset.name}")
+                if self.buttons and asset.name not in [
+                    a.unselected_image_asset for a in self.buttons
+                ]:
+                    raise ValidationError(f"Asset name not in buttons: {asset.name}")
+                if self.images and asset.name not in [
+                    a.image_asset for a in self.images
+                ]:
+                    raise ValidationError(f"Asset name not in images: {asset.name}")
+
+        for a in self.assets:
+            if not Path(a.path).is_file():
+                raise ValidationError(f"Asset path does not exist: {a.path}")
+
+        duplicate_buttons = [
+            button for button, count in Counter(self.buttons).items() if count > 1
```

Comment:
Remember that uniqueness should be on the _name_, not on the button itself. `Counter` (along with sets and dicts) uses [hashing](https://docs.python.org/3/glossary.html#term-hashable) to determine uniqueness of objects. If two buttons have the same name but differ in any other field, they will have different hashes and therefore this validation will not catch the duplicate entry.

This will also fail to validate if the name is not unique across different object types - a button could have the same name as an image, which is not what we want.

Instead consider just collecting a set of the names of the objects:
```python
names = collections.Counter()
names.update(b.name for b in self.buttons)
# repeat for assets and images

duplicates = {name for name, count in names.values() if count > 1}
```